### DISCUSSION TITLE
Why does athena.to_iceberg only support insert?

### DISCUSSION THREAD

**Discussion by brendan-cook-87** [aws/aws-sdk-pandas#2522] [Q&A] [answered]
Basically the question above.
I started out trying to process some events in a lambda (reading a jsonl file  from a S3 object notification). And for testing purposes I processed the same file twice and obviously that duplicates the event.

Is there a reason it doesn't support a merge into strategy where we specify the merge_cols?
Generate a query like

```
MERGE INTO table USING temp_table
    ON (table.merge_col = temp_table.merge_col)

    WHEN MATCHED THEN UPDATE
        SET col1 = temp_table.col1 ...

    WHEN NOT MATCHED THEN INSERT (col1 ...)
         VALUES(temp_table.col1 ...)
```

**Comment by kukushking** [comment, marked as answer]
Closing as MERGE INTO supported since [3.5.0](https://github.com/aws/aws-sdk-pandas/releases/tag/3.5.0).
